### PR TITLE
feat: add other os build workflow

### LINKED ISSUE #3302: build cxx binary/so for lower glibc version or other os
If released binary/so runs on lower glibc host, it'll get error:
```
libstdc++.so.6: version `GLIBCXX_3.4.19' not found
```

- [ ] it's better to fix the exception throw in LibraryLoader, so that we can see the better error msg.

We check the host system lib by script
```
#!/bin/bash
set -x
cat /etc/redhat-release
ldd --version
strings /lib64/libc.so.6 | grep ^GLIBC_
strings /usr/lib64/libstdc++.so.6 |grep ^GLIBC
```

For example, if we want run binary/so on glibc 2.12 and libstdc++.so.6.0.13(GLIBCXX_3.4.13), we can build it on a old centos image.

Build steps:
1. use old image, `ghcr.io/4paradigm/centos6_gcc7_hybridsql`
2. yum install -y bison bison-devel byacc cppunit-devel patchdevtoolset-8-gcc devtoolset-8-gcc-c++, install build tools.
3. create `/etc/os-release` for zetasql
4. rocksdb don't build sst_dump&ldb, or add -lrt
5. Download https://github.com/sub-mod/bazel-builds/releases for zetasql, build_zetasql_parser.sh直接关了bazel test? add -lrt
6. add func in script(centos6 coreutils doesn't have `realpath`)
```
realpath () { 
[[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}" 
}
```
Better to add a new Dockerfile? The repo is private and we should modify zetasql build scripts, so add a script for patch is better. If we want a new Dockerfile, thirdparty build scripts need update.

We can test the built files, e.g. `strings libsql_jsdk.so | grep ^GLIBC`

### PR DESCRIPTION
**Pull request by vagetablechicken**
close #3302 
workflow for centos6, macos 10/11, just dispatch trigger
build script for centos6

### CODE REVIEW

File: .github/workflows/other-os-build.yml [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    if: ${{ github.event.inputs.OS == 'centos6' }}
+    runs-on: ubuntu-latest
+    env:
+      OS: linux
+      SQL_JAVASDK_ENABLE: ${{ github.event.inputs.JAVA_SDK }}
+      SQL_PYSDK_ENABLE: ${{ github.event.inputs.PYTHON_SDK }} # python whl will be built when make, no prerequirement
+      TESTING_ENABLE: OFF
+      NPROC: 8
+    steps:
+      - uses: actions/checkout@v3
```
Comment (aceforeverd) [inline .github/workflows/other-os-build.yml:41]: give the option  to build upon release tag, and tweak release artifact name

File: steps/centos6_build.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
```
Comment (dl239) [inline steps/centos6_build.sh:1]: add copyright

File: .github/workflows/other-os-build.yml (hunk 1 above)
Comment (vagetablechicken) [reply .github/workflows/other-os-build.yml:41]: You mean a option `VERSION` to let user define the release version?

File: steps/centos6_build.sh (hunk 2 above)
Comment (vagetablechicken) [reply steps/centos6_build.sh:1]: done

**Review by dl239 — APPROVED** [review]

File: .github/workflows/other-os-build.yml (hunk 1 above)
Comment (vagetablechicken) [reply .github/workflows/other-os-build.yml:41]: done

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**